### ISSUE TITLE
IntelliSense error should not be marked

### ISSUE THREAD

**Issue by Freekjay** [microsoft/azuredatastudio#21771]
Issue Type: <b>Bug</b>

- Azure Data Studio Version: 1.40.2
- OS Version: Windows 11
- SQL Database Projects Version: 0.21.0

Steps to Reproduce:

1. Add the (preview) extension SQL Database Projects
2. Use an already existing SQL Database Project or create a new one
3. Edit a stored procedure in the project **which already exists**
4. Notice that it gives the error message "There is already an object named <name of stored procedure>"

Extension version: 0.21.0
Azure Data Studio version: azuredatastudio 1.40.2 (661384637db384fe5d4e6224069adbe708580b16, 2022-12-22T22:17:02.658Z)
OS version: Windows_NT x64 10.0.22621
Restricted Mode: No
Preview Features: Enabled

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz (12 x 2592)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.75GB (7.11GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by Freekjay** [comment]
![image](https://user-images.githubusercontent.com/5197103/215293501-99b1ca7d-447f-40be-b671-1ec0ea5dd11b.png)

**Comment by Charles-Gagnon** [comment]
Does the error continue to appear if you run the `Refresh Intellisense Cache` command?

**Comment by Freekjay** [comment]
Yes. That's when it actually starts appearing because it can see the stored procedure on the server already existing.

**Comment by Charles-Gagnon** [comment]
Oh, I misunderstood that. So you're saying you think it should not show up as an error? But the object does exist so this is currently expected behavior.

If you add a `DROP PROCEDURE IF EXISTS` at the top of this file then you shouldn't see the error - which is good as long as you're ok with this script always ensuring that it will recreate the function if it exists already for some reason. 

```sql
DROP PROCEDURE IF EXISTS foo
GO

CREATE PROCEDURE foo AS SELECT 1
```

@kisantia @llali @Benjin Is this the generally accepted way to do this or is there some other feature I'm missing?

**Comment by kisantia** [comment]
The scripts in a sql project are for creating all the objects on an empty database. To update an existing database, use the "publish" option should be used to generate the delta script or publish the changes directly to the database, rather than running the scripts in the sql project directly.
![image](https://user-images.githubusercontent.com/31145923/215553945-93f806d0-32d5-45db-b189-cb57cb19f358.png)

**Comment by Charles-Gagnon** [comment]
It's kind of quirky behavior though where you can have these sql files and as long as the database doesn't exist on the server then it'll seem fine, but as soon as you publish it then all them will show errors though (since the database + objects exist there now).

The editors don't have any idea about the context that this is supposed to be ran on a "new" database - it just sees what the current state is.

Does SSDT have this same behavior? Does it even connect to the actual server for parsing or just parse the local files separately?

**Comment by kisantia** [comment]
The sql projects extension in ADS opens .sql files in a project using the `vscode.open` command, but it's only intended for users to view and manually edit the files, not to run them against the database themselves. "Publish" or "Schema Compare" should be used for pushing any changes from sql projects to dbs.

This is what it looks like when opening a .sql file in a project in SSDT - no option to connect and run
![image](https://user-images.githubusercontent.com/31145923/215614685-39c12b6e-2af2-41f7-bcc0-d32a6e25d9e0.png)

**Comment by Benjin** [comment]
Perhaps we can add a "headless" mode to the open file context, where it's not trying to compare against any schema?

**Comment by alanrenmsft** [comment]
@Charles-Gagnon @kisantia could you please find a proper milestone for this issue?
